src/v/cloud_storage/remote.h

```diff
@@ -285,6 +285,20 @@ class remote : public ss::peering_sharded_service<remote> {
       std::optional<cloud_storage_clients::http_byte_range> byte_range
       = std::nullopt);
 
+    ss::future<upload_result> upload_file(
+      const cloud_storage_clients::bucket_name& bucket,
+      const remote_segment_path& remote_path,
+      const ss::file& file,
+      retry_chain_node& parent,
+      lazy_abort_source& lazy_abort_source,
+      const cloud_storage_clients::object_tag_formatter& tags
+      = default_segment_tags);
+    ss::future<download_result> download_file(
+      const cloud_storage_clients::bucket_name& bucket,
+      const remote_segment_path& remote_path,
+      const try_consume_stream& cons_str,
+      retry_chain_node& parent);
```

Comment:
Done, made a templatized download/upload_stream; callers can pass in their own metrics (though down the line, maybe we should just pass in the metric counters and measurers, instead of these functors)